Purchases/Public/RCPurchases.m

```diff
@@ -405,7 +405,7 @@ - (void)purchaserInfoWithCompletionBlock:(RCReceivePurchaserInfoBlock)completion
     RCPurchaserInfo *infoFromCache = [self readPurchaserInfoFromCache];
     if (infoFromCache) {
         RCDebugLog(@"Vending purchaserInfo from cache");
-        CALL_IF_SET(completion, infoFromCache, nil);
+        CALL_AND_DISPATCH_IF_SET(completion, infoFromCache, nil);
```

Comment:
We don't really have to dispatch here since there is no thread change. We are calling completion from the same thread it's called. What do you think?